src/omp/backend.f90

```diff
@@ -379,18 +412,91 @@ subroutine vecadd_omp(self, a, x, b, y)
     class(field_t), intent(in) :: x
     real(dp), intent(in) :: b
     class(field_t), intent(inout) :: y
+    integer, dimension(3) :: dims
+    integer :: i, j, k, ii
+
+    integer :: nvec, remstart
+
+    if ((x%dir /= y%dir) .or. (x%data_loc /= y%data_loc)) then
```

Comment:
I wouldn't do a sanity check for the `data_loc` here because for example time integrator arrays don't need a `data_loc`, but the arrays they're summed over sometimes have a `data_loc`. I think its reasonable to leave this to the user. We can even use `vecadd` to obtain crude interpolations later on if we need using two arrays with different `data_loc`s.

Reply:
Done